internal/fetcher/fetcher.go

```diff
@@ -75,3 +77,27 @@ func readHTTP(ctx context.Context, u *url.URL) ([]byte, error) {
 	}()
 	return io.ReadAll(resp.Body)
 }
+
+func BinaryFetcher(f client.Fetcher) func(context.Context, ftlog.FirmwareRelease) ([]byte, []byte, error) {
+	return func(ctx context.Context, r ftlog.FirmwareRelease) ([]byte, []byte, error) {
+		p, err := update.BinaryPath(r)
+		if err != nil {
+			return nil, nil, fmt.Errorf("BinaryPath: %v", err)
+		}
+		klog.Infof("Fetching %v bin from %q", r.Component, p)
+		var b, s []byte
+		if b, err = f(ctx, p); err != nil {
+			return nil, nil, fmt.Errorf("failed to get %v binary from %q: %v", r.Component, p, err)
+		}
+		if len(r.HAB.SignatureDigestSha256) != 0 {
+			if p, err = update.HABSignaturePath(r); err != nil {
+				return nil, nil, fmt.Errorf("HABSignaturePath: %v", err)
+			}
+			b, err = f(ctx, p)
```

Comment:
```suggestion
			s, err = f(ctx, p)
```

I think documenting the return value names would be useful. We should probably set up a fake server implementation that this could query for testing, but that's probably beyond the scope of this PR.

Reply:
Done